packages/cx-diagrams/src/StraightLine.js

```diff
@@ -88,6 +96,36 @@ function circleIntersection(t, l, b, r, radius) {
    };
 }
 
+function rhombusIntersection(x1, y1, x2, y2, halfw, halfh) {
+   if (x1 == x2) {
+      return {
+         x: x1,
+         y: y2 > y1 ? y1 + halfh : y1 - halfh,
+      };
+   }
+
+   if (y1 == y2) {
+      return {
+         y: y1,
+         x: x2 > x1 ? x1 + halfw : x1 - halfw,
+      };
+   }
+
+   const v1y = y1 > y2 ? y1 - halfh : y1 + halfh;
+   const v1x = x1;
+
+   const v2y = y1;
+   const v2x = x1 > x2 ? x1 - halfw : x1 + halfw;
+
+   const { x, y } = getLinesIntersectionPoint(v1x, v1y, v2x, v2y, x1, y1, x2, y2);
+
+   if (x === undefined || y === undefined) {
+      throw new Error("Could not find intersection point.");
+   }
```

Comment:
Instead of throwing an error, it would be better to return the center point in case there is no intersection.